### DISCUSSION TITLE
A research for generating PR checklists in Pull Request Template

### DISCUSSION THREAD

**Discussion by baishuotong** [microsoft/monaco-editor#4275] [Q&A]
Dear maintainers:
Our team is currently developing a tool for automatically generating PR checklists in Pull Request Template. The tool generates a set of PR rules based on the contribution guidelines in your project, which external contributors may be expected to adhere to. Below are the results of our generation:

**Test**: Run tests using various package.
**Documentation**: Create necessary files for new language.
**commit message**: Use commit in vscodeRef in package.json

Also, we suggest you add some checklist notes about **Style**, **Branch**,**title**, **description**, **Documentation** , **contributing**, and **cla**, which are also important for submitting PRs according to our analysis.

We hope you and your team can provide feedback on our generated output.
Best regards.

**Comment by hediet** [comment]
Unfortunately, the generated results are not very helpful.
